### PR TITLE
Implement trek check

### LINKED ISSUE #16: Validate that all files in the `migration` folder follow the naming convention
As part of the `trek check` subcommand.

Also validate this on the `generate` subcommand. Exit immediatly with an error when validation fails.

### LINKED ISSUE #17: Validate dbm file
Roles and the database should always have `sql-disabled="true"`, because we create those with trek itself.

### LINKED ISSUE #41: Validate testdata after generating migration file

### LINKED ISSUE #47: Validate that a migration number is only used once
When rebasing a migration number might be used twice, then it breaks.

### LINKED ISSUE #52: Implement `trek check`
Should check all of
https://github.com/stack11/trek/issues/47
https://github.com/stack11/trek/issues/41
https://github.com/stack11/trek/issues/17
https://github.com/stack11/trek/issues/16
https://github.com/stack11/trek/issues/55

And add a hook so users can do custom validation (like checking if a readonly user has access to everything).

Checking should also be performed by default every time after generating a migration.

### LINKED ISSUE #55: Validate that all templates are up to date

### PR DESCRIPTION
**Pull request by provokateurin**
closes https://github.com/stack11/trek/issues/55  
closes https://github.com/stack11/trek/issues/52  
closes https://github.com/stack11/trek/issues/47  
closes https://github.com/stack11/trek/issues/41  
closes https://github.com/stack11/trek/issues/17  
closes https://github.com/stack11/trek/issues/16

### CODE REVIEW

File: cmd/check.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+			_ = conn.Close(ctx)
+		}
+		if postgres != nil {
+			_ = postgres.Stop()
+		}
+	}()
+	if err != nil {
+		return fmt.Errorf("failed to setup database: %w", err)
+	}
+	dsn = fmt.Sprintf("%s?sslmode=disable", dsn)
```
Comment (dmeijboom) [inline cmd/check.go:84]: Why is `sslmode` disabled?

File: cmd/check.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+	}
+
+	hookOptions := &internal.HookOptions{
+		Env: map[string]string{
+			"TREK_POSTGRES_HOST":     "localhost",
+			"TREK_POSTGRES_PORT":     "5434",
+			"TREK_POSTGRES_USER":     "postgres",
+			"TREK_POSTGRES_PASSWORD": "postgres",
+			"TREK_POSTGRES_DATABASE": "postgres",
+			"TREK_POSTGRES_SSLMODE":  "disable",
```
Comment (dmeijboom) [inline cmd/check.go:112]: Why are there hardcoded options?

File: cmd/check.go [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+
+	err = xml.Unmarshal(m, &model)
+	if err != nil {
+		return fmt.Errorf("failed to parse model: %w", err)
+	}
+
+	modelRoles := map[string]struct{}{}
+	for _, role := range model.Roles {
+		if !role.SQLDisabled {
+			//nolint:goerr113
```
Comment (dmeijboom) [inline cmd/check.go:180]: Why is this required? (the nolint rules)

**Review by dmeijboom — CHANGES_REQUESTED** [review]
Not for now but I do wonder if it makes sense to have a more high-level design at some point. Seems like there is a lot of logic hidden in the CLI

File: cmd/check.go (hunk 2 above)
Comment (jld3103) [reply cmd/check.go:112]: trek uses embedded databases for diffing and checking and those are just the default settings for the check database. they need to be hardcoded that way.

File: cmd/check.go (hunk 1 above)
Comment (jld3103) [reply cmd/check.go:84]: the embedded database doesn't have ssl enabled

File: cmd/check.go (hunk 3 above)
Comment (jld3103) [reply cmd/check.go:180]: this specific linter has this behaviour which we don't want, but other behaviour that we want (like error comparison using `errors.Is(...)`. that's why it's ignored.

**Review by dmeijboom — APPROVED** [review]
